Data Protection Firewall: sharing -> bi-directional sharing (#449)
MicrosoftDocs/powerquery-docs: powerquery-docs/data-privacy-firewall.md

* A partition may either access compatible data sources, or reference other partitions, but not both.

> In order for your single-partition-but-two-data-sources query to be allowed to run, its two data sources must be "compatible". In other words, it needs to be okay for data to be shared bi-directionally between them (that is, it needs to be okay for data from _either_ source to be folded into the other). Effectively, this means that the privacy levels of both sources need to be Public, or both be Organizational, as these are the only two combinations that allow sharing in both directions. If the sources are both marked Private, or one is marked Public and one is marked Organizational, or they're marked using some other combination of privacy levels, then bi-directional sharing is not allowable, so it's not safe for them to both be evaluated in the same partition. Doing so would mean unsafe data leakage could occur (due to folding), and the Firewall would have no way to prevent it.

Reviewer: ```suggestion
In order for your single-partition-but-two-data-sources query to be allowed to run, its two data sources must be "compatible". In other words, it needs to be okay for data to be shared bidirectionally between them. This means that the privacy levels of both sources need to be Public, or both be Organizational, since these are the only two combinations that allow sharing in both directions. If both sources are marked Private, or one is marked Public and one is marked Organizational, or they're marked using some other combination of privacy levels, then bidirectional sharing isn't allowed, and it's thus not safe for them to both be evaluated in the same partition. Doing so would mean unsafe data leakage could occur (due to folding), and the Firewall would have no way to prevent it.
```

Author: Thanks, @ehrenMSFT!
